src/PostgREST/Middleware.hs

```diff
@@ -83,6 +89,16 @@ runPgLocals conf claims app req jsonDbS = do
     setConfigLocal :: ByteString -> (ByteString, ByteString) -> H.Snippet
     setConfigLocal prefix (k, v) =
       "set_config(" <> unknownEncoder (prefix <> k) <> ", " <> unknownEncoder v <> ", true)"
+    -- | Starting from PostgreSQL v14, some characters are not allowed for config names (mostly affecting headers with "-").
+    -- | A JSON format string is used to avoid this problem. See https://github.com/PostgREST/postgrest/issues/1857
+    setGuc :: ByteString -> ByteString -> [(ByteString, ByteString)] -> [H.Snippet]
+    setGuc prefixString prefixJson keyVals
```

Comment:
I think this should be an independent function(outside of the `where`), also maybe rename the args like:

```suggestion
    setGuc prefixPgVerLt14 prefixPgVerGte14 keyVals
```

And a comment mentioning that the prefixes change between versions?

Reply:
`setGuc` uses the `setConfigLocal` function, in this case is it okay to also take that function out of the `where` too?

Reply:
Yeah, should be ok This `where` part seems too long now.